domain/consensus/processes/dagtraversalmanager/selected_child_iterator.go

```diff
@@ -9,21 +9,19 @@ import (
 type selectedChildIterator struct {
 	databaseContext    model.DBReader
 	dagTopologyManager model.DAGTopologyManager
-	highHash           *externalapi.DomainHash
-	current            *externalapi.DomainHash
+
+	reachabilityDataStore model.ReachabilityDataStore
+	highHash              *externalapi.DomainHash
+	current               *externalapi.DomainHash
 }
 
 func (s *selectedChildIterator) Next() bool {
-	children, err := s.dagTopologyManager.Children(s.current)
+	data, err := s.reachabilityDataStore.ReachabilityData(s.databaseContext, s.current)
 	if err != nil {
 		panic(err)
 	}
 
-	for _, child := range children {
-		if *child == *model.VirtualBlockHash {
-			continue
-		}
-
+	for _, child := range data.TreeNode.Children {
```

Comment:
I think this loop can be replaced with a binary search - using the fact that tree children are ordered by interval 
(in previous versions of the codebase this logic was used in a method called findAncestorOfThisAmongChildrenOfOther - discuss with @svarogg @stasatdaglabs  )